systemtest/src/test/java/io/strimzi/systemtest/operators/PodSetIsolatedST.java

```diff
@@ -0,0 +1,117 @@
+ * they should be replacement for StatefulSets in the future.
+ */
+@Tag(REGRESSION)
+@IsolatedSuite
+public class PodSetIsolatedST extends AbstractST {
+
+    private static final Logger LOGGER = LogManager.getLogger(PodSetIsolatedST.class);
+
+    private static final List<EnvVar> INITIAL_ENV_VARS = Collections.singletonList(new EnvVar(Environment.STRIMZI_FEATURE_GATES_ENV, Constants.USE_STRIMZI_POD_SET, null));
+
+    @IsolatedTest("We are changing CO env variables in this test")
+    void testPodSetOnlyReconciliation(ExtensionContext extensionContext) {
+        final TestStorage testStorage = new TestStorage(extensionContext);
+        final Map<String, String> coPod = DeploymentUtils.depSnapshot(INFRA_NAMESPACE, STRIMZI_DEPLOYMENT_NAME);
+        final int replicas = 3;
+        final int probeTimeoutSeconds = 6;
+
+        EnvVar reconciliationEnv = new EnvVar(Environment.STRIMZI_POD_SET_RECONCILIATION_ONLY_ENV, "true", null);
+        List<EnvVar> envVars = kubeClient().getDeployment(INFRA_NAMESPACE, Constants.STRIMZI_DEPLOYMENT_NAME).getSpec().getTemplate().getSpec().getContainers().get(0).getEnv();
+        envVars.addAll(INITIAL_ENV_VARS);
+        envVars.add(reconciliationEnv);
+
+        resourceManager.createResource(extensionContext, KafkaTemplates.kafkaPersistent(testStorage.getClusterName(), replicas).build());
+
+        LOGGER.info("Changing {} to 'true', so only SPS will be reconciled", Environment.STRIMZI_POD_SET_RECONCILIATION_ONLY_ENV);
+
+        DeploymentResource.replaceDeployment(Constants.STRIMZI_DEPLOYMENT_NAME,
+            coDep -> coDep.getSpec().getTemplate().getSpec().getContainers().get(0).setEnv(envVars), INFRA_NAMESPACE);
+
+        DeploymentUtils.waitTillDepHasRolled(INFRA_NAMESPACE, STRIMZI_DEPLOYMENT_NAME, 1, coPod);
+
+        Map<String, String> kafkaPods = PodUtils.podSnapshot(testStorage.getNamespaceName(), testStorage.getKafkaSelector());
+        Map<String, String> zkPods = PodUtils.podSnapshot(testStorage.getNamespaceName(), testStorage.getZookeeperSelector());
+
+        LOGGER.info("Changing Kafka resource configuration, the pods should not be rolled");
+
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getClusterName(),
+            kafka -> kafka.getSpec().getZookeeper().setReadinessProbe(new ProbeBuilder().withTimeoutSeconds(probeTimeoutSeconds).build()), testStorage.getNamespaceName());
+        RollingUpdateUtils.waitForNoRollingUpdate(testStorage.getNamespaceName(), testStorage.getZookeeperSelector(), zkPods);
+
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getClusterName(),
+            kafka -> kafka.getSpec().getKafka().setReadinessProbe(new ProbeBuilder().withTimeoutSeconds(probeTimeoutSeconds).build()), testStorage.getNamespaceName());
+        RollingUpdateUtils.waitForNoRollingUpdate(testStorage.getNamespaceName(), testStorage.getKafkaSelector(), kafkaPods);
```

Comment:
Any chance you can change both at once and wait once? That would save some execution time and the value will be the same.

Reply:
Makes sense, good catch :) thanks